### ISSUE TITLE
layout:decorator attribute in version 2.0.0 isn't removed during processing

### ISSUE THREAD

**Issue by cja769** [ultraq/thymeleaf-layout-dialect#125]
Even though the migration notes say that the `layout:decorator` attribute should work in the new 2.0.0 release I found that it does not. It seems that the reason is this piece of code in `DecorateProcessor.doProcess`:
```
// Remove the layout:decorate attribute for cases when the root element is
// also a potential fragment
def rootElement = model.first()
if (rootElement.hasAttribute(dialectPrefix, PROCESSOR_NAME)) {
    rootElement = context.modelFactory.removeAttribute(rootElement, dialectPrefix, PROCESSOR_NAME)
    model.replace(0, rootElement)
}
```
`PROCESSOR_NAME` is set in `DecorateProcessor` as `decorate` and `PROCESSOR_NAME` is set in `DecoratorProcessor` as `decorator` but `DecoratorProcessor` uses `super.doProcess` (i.e. DecorateProcessor.doProcess) so the conditional resolves as false. Since that happens, later down the execution path when the root element attributes in the template with the `layout:decorator` gets merged onto the root element element of the decorating template, the `layout:decorator` attribute gets added to the `<html>` tag.

This wouldn't be a problem except that the `layout:decorator` attribute gets processed again for the html tag and now the template has two `layout:decorator`s which makes the following fail in `DecorateProcessor.doProcess`:
```
// Ensure the decorate attribute is in the root element of the document
if (context.elementStack.size() != 1) {
    throw new IllegalArgumentException('layout:decorate/data-layout-decorate must appear in the root element of your template')
}
```

**Comment by cja769** [comment]
Closing issue because this is fixed by commit f38e066006af1836cb21e506623c0247dac031c5

**Comment by ultraq** [comment]
FYI, that fix/commit will be a part of 2.0.1, which is available in a SNAPSHOT form right now, but I'm also aiming to get a proper 2.0.1 release done before this week is over.
